### PR TITLE
HPCC-16458 Spray regression test cases failed after drop zone restriction reactivated.

### PR DESCRIPTION
**Pull request by AttilaVamos**
Rewrite test cases to avoid spray from /testing/regress/download:
1. generate simple dataset
2. despray it to default drop zone
3. (fixed, variable) spray it back from the default drop zone

Signed-off-by: Attila.Vamos attila.vamos@gmail.com

### CODE REVIEW

**Comment by AttilaVamos** [comment]
Actually it is still WIP.

**Comment by AttilaVamos** [comment]
@ghalliday please review.

**Comment by AttilaVamos** [comment]
@ghalliday it is modified. It compares result dataset inside the ECL and outputs 'Passed' or 'Fail' instead. Please review.

File: testing/regress/ecl/spray_test.ecl [hunk 1]
```diff
@@ -24,25 +24,119 @@ import Std.File AS FileServices;
...
+
+Layout_Person := RECORD
+  STRING3  name;
+  UNSIGNED2 age;
+  BOOLEAN good;
+END;
+
+sprayPrepFileName := '~::spray_prep';
+desprayOutFileName := '/var/lib/HPCCSystems/mydropzone/spray_input';
+sprayOutFileName := '~::spray_test';
```
Comment (ghalliday) [inline testing/regress/ecl/spray_test.ecl]: I think other files generated by the regression suite are in the regress (check that!) scope, this should be as well.

File: testing/regress/ecl/spray_test.ecl [hunk 2]
```diff
@@ -24,25 +24,119 @@ import Std.File AS FileServices;
...
+
+unsigned VERBOSE := 0;
+
+Layout_Person := RECORD
+  STRING3  name;
+  UNSIGNED2 age;
+  BOOLEAN good;
+END;
+
+sprayPrepFileName := '~::spray_prep';
```
Comment (ghalliday) [inline testing/regress/ecl/spray_test.ecl]: same comment as below

File: testing/regress/ecl/spray_test.ecl [hunk 3]
```diff
@@ -24,25 +24,119 @@ import Std.File AS FileServices;
...
+#if (VERBOSE = 1)
+    sprayOut := output(c2);
+#else
+    sprayOut := output(c2, {result});
+#end
+
+ds := DATASET(sprayOutFileName, Layout_Person, csv);
+
+string compareDatasets(dataset(Layout_Person) ds1, dataset(Layout_Person) ds2) := FUNCTION
+   boolean result := (0 = COUNT(JOIN(ds1, ds2, left.name=right.name, FULL ONLY, LOCAL)));
```
Comment (ghalliday) [inline testing/regress/ecl/spray_test.ecl]: Shouldn't really have LOCAL here.  It is ignored by roxie and hthor, and would be wrong for thor.

File: testing/regress/ecl/spray_test.ecl [hunk 4]
```diff
@@ -24,25 +24,119 @@ import Std.File AS FileServices;
...
+import ^ as root;
+
+boolean sprayFixed := #IFDEFINED(root.sprayFixed, true);
+
+unsigned VERBOSE := 0;
+
+Layout_Person := RECORD
+  STRING3  name;
+  UNSIGNED2 age;
+  BOOLEAN good;
```
Comment (ghalliday) [inline testing/regress/ecl/spray_test.ecl]: I was concerned that this would generate true/false in the output, but that doesn't seem to be the case.

**Review by ghalliday — CHANGES_REQUESTED** [review]
@AttilaVamos looks ok - one comment on the scope of the files being produced.

File: testing/regress/ecl/spray_test.ecl (hunk 4 above)
Comment (AttilaVamos) [inline testing/regress/ecl/spray_test.ecl]: Ok

File: testing/regress/ecl/spray_test.ecl (hunk 2 above)
Comment (AttilaVamos) [inline testing/regress/ecl/spray_test.ecl]: Ok

File: testing/regress/ecl/spray_test.ecl (hunk 1 above)
Comment (AttilaVamos) [inline testing/regress/ecl/spray_test.ecl]: Ok

**Comment by AttilaVamos** [comment]
@ghalliday updated, please check.

**Review by ghalliday — APPROVED** [review]

**Pull request merged**